File: multi-module-project/movie-api/src/main/java/com/dpflsy/movie/controller/MovieController.java
package com.dpflsy.movie.controller;

import com.dpflsy.common.dto.MovieResponse;
import com.dpflsy.movie.service.MovieService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class MovieController {
    private final MovieService movieService;

    @GetMapping("/api/movies/now-playing")
    public List<MovieResponse> getNowPlayingMovies() {
        return movieService.getNowPlayingMovies();
    }
}

File: movie/api/src/main/kotlin/com/example/movie/movie/controller/MovieController.kt
package com.example.movie.movie.controller

import com.example.movie.movie.dto.MovieResponse
import com.example.movie.service.NowPlayingMoviesUseCase
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/movies")
class MovieController(
    private val nowPlayingMoviesUseCase: NowPlayingMoviesUseCase
) {
    @GetMapping("/now-playing")
    fun getNowPlaying(): List<MovieResponse> {
        val movieScreenings = nowPlayingMoviesUseCase.getNowPlayingMovies()
        return movieScreenings.map { (movie, screenings) ->
            MovieResponse.from(movie, screenings)
        }
    }
}